### PR TITLE
data_override_init logic updates

### PR DESCRIPTION
**Pull request by bensonr**
Fixed an incorrect if comparison (int ? real) in simple coupler.
Added same if-test logic for data_table to the SHiELD coupler plus removing some unused variables and logic.

There are 8 extra commits here because of a previous squash merge of this branch into main.

### CODE REVIEW

File: SHiELD/coupler_main.F90 [hunk 1]
```diff
@@ -388,12 +385,16 @@ subroutine coupler_init
 
     call print_memuse_stats('after atmos model init')
 
-    call mpp_get_global_domain(Atm%Domain, xsize=gnlon, ysize=gnlat)
-    allocate ( glon_bnd(gnlon+1,gnlat+1), glat_bnd(gnlon+1,gnlat+1) )
-    call mpp_global_field(Atm%Domain, Atm%lon_bnd, glon_bnd, position=CORNER)
-    call mpp_global_field(Atm%Domain, Atm%lat_bnd, glat_bnd, position=CORNER)
-
-    if (.NOT.Atm%bounded_domain) call data_override_init (Atm_domain_in  = Atm%domain)
+    if (file_exists('data_table') .and. .NOT.Atm%bounded_domain) then
```
Comment (thomas-robinson) [inline SHiELD/coupler_main.F90:388]: Does this need to work with the data_table.yaml?  Does this coupler version need to be updated @uramirez8707 ?

**Review by thomas-robinson — COMMENTED** [review]
Need to check with @uramirez8707 that this doesn't need to include logic for a data_table.yaml file.  Also @GFDL-Eric should make sure this logic is the same as the other couplers.  I vaguely remember that we ran into issues with something similar in the doubly periodic.

**Comment by GFDL-Eric** [comment]
@thomas-robinson yes, this logic is in the simple coupler, although it is not in the full coupler. I believe this was because an empty table and/or a non-existent table was more likely in models that use the simple coupler than those that use the full coupler. We will likely have to add similar logic for data_table.yaml at some point in the future, I suppose we could do it now just to cover our bases?

**Comment by uramirez8707** [comment]
I think that this should be handled in data_override and not in coupler. 

This was happening because a user wanted to use data_override, but data_override_init was not being called with the domain argument so nothing was happening. We then added the data_override_init call, but other models (doubly_periodic) were crashing because the grid_spec file was not present. 

We should add the logic here:
https://github.com/NOAA-GFDL/FMS/blob/7b78a73a5ba7acf5d3d932ecfe081e5040e2c778/data_override/data_override.F90#L263

```F90
if (table_size .eq. 0) then 
  call mpp_error(NOTE, "data_table is empty, not doing any data_overrides")
  return
endif
```
where table_size is set when reading the data_table or data_table.yaml

This will be easier to maintain ...

**Comment by GFDL-Eric** [comment]
@uramirez8707 I support this, but we have to be cognizant of the current state of data_override, in which the code will error if the data_table or data_table.yaml file is missing. Is there a reason for this behavior? My impression is that calling data_override_init should just be something that happens in all the couplers (one time for each of the models that are being coupled [atm, lnd, etc.]). If that's how we want to proceed, we shouldn't prohibit the user from not having a data_table if there's nothing to be overridden. Essentially, I want to get rid of https://github.com/NOAA-GFDL/FMS/blob/7b78a73a5ba7acf5d3d932ecfe081e5040e2c778/data_override/data_override.F90#L370 and replace it with something that will just return out of the routine while establishing table_size as 0. Does anyone have thoughts on this?

**Comment by bensonr** [comment]
Different fix is being implemented

**Pull request closed without merging**